### PR TITLE
Add sc_pid to SCU subdevice sysfs to support PS kernel stat gathering

### PR DESCRIPTION
**Pull request by jeffli-xilinx**
#### Problem solved by the commit
Allows query of PS kernel pid

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
None

#### How problem was solved, alternative solutions (if any) and why they were rejected
Expose pid through sysfs node on SCU platform subdevice

#### Risks (if any) associated the changes in the commit
None

#### What has been tested and how, request additional testing if necessary
PS kernel validate tests

#### Documentation impact (if any)
Example sysfs node:
/sys/devices/platform/ert_hw/SCU.4.auto/sc_pid

### CODE REVIEW

**Review by rozumx — COMMENTED** [review]
Jeff,
Could you update the PR description with an example directory structure.

File: src/runtime_src/core/edge/drm/zocl/scu.c [hunk 1]
```diff
@@ -92,11 +92,33 @@ stat_show(struct device *dev, struct device_attribute *attr, char *buf)
 }
 static DEVICE_ATTR_RO(stat);
 
+ssize_t show_sc_pid(u32 sc_pid, char *buf)
+{
+	ssize_t sz = 0;
+	char *fmt = "%u\n";
```
Comment (dbenusov-xilinx) [inline src/runtime_src/core/edge/drm/zocl/scu.c:98]:
```suggestion
	char *fmt = "PID:%u\n";
```
To match all the other files would it be a good idea to have the format like this? Even if it is one element.

File: src/runtime_src/core/edge/drm/zocl/scu.c [hunk 2]
```diff
@@ -92,11 +92,33 @@ stat_show(struct device *dev, struct device_attribute *attr, char *buf)
...
+
+static ssize_t
+sc_pid_show(struct device *dev, struct device_attribute *attr, char *buf)
+{
+	struct platform_device *pdev = to_platform_device(dev);
+	struct zocl_scu *scu = platform_get_drvdata(pdev);
+
+	return show_sc_pid(scu->sc_pid, buf);
+}
+static DEVICE_ATTR_RO(sc_pid);
```
Comment (dbenusov-xilinx) [inline src/runtime_src/core/edge/drm/zocl/scu.c:114]:
```suggestion
static DEVICE_ATTR_RO(status);
```
If there is every additional data would we want to create a different node? We could place it all into one attribute file to reduce the amount of files to access. Renaming the attribute to `status` mimics the behavior of the `/proc/` entries. The PID of the process in `/proc/` is also stored in the `status` attribute so the logic matches up. Any additional information we would ever need to add could be easily parsed into the PS kernel I am developing by adding another row of data.

**Review by dbenusov-xilinx — CHANGES_REQUESTED** [review]

**Review by chvamshi-xilinx — APPROVED** [review]

**Review by dbenusov-xilinx — APPROVED** [review]
Thank you!

**Pull request merged**